### DISCUSSION TITLE
原始数据存在换行，如何展示

### DISCUSSION THREAD

**Discussion by guogeer** [clickvisual/clickvisual#1046] [Q&A]
![image](https://github.com/clickvisual/clickvisual/assets/5010851/7091033e-91fc-4db7-930c-48ef8c5900ec)
现在的效果展示可读性差

**Comment by kl7sn** [comment]
建议用结构化日志，例如JSON

**Reply by guogeer** [reply]
感谢回复！这个是错误日志堆栈，如何结构化？转化成json数组？

**Comment by kl7sn** [comment]
确实没有考虑到直接显示字符串的情况，如果堆栈是日志的一个部分，显示效果如下
![image](https://github.com/clickvisual/clickvisual/assets/2037801/78537c85-066a-4930-8a63-2dbfe42e6649)

**Comment by guogeer** [comment]
感谢。上图中stack是个json 数组？

**Reply by kl7sn** [reply]
_raw_log_ 部分是 JSON，stack 是里面的一个字段，内容为 string
